### DISCUSSION TITLE
Sell app that work with llama

### DISCUSSION THREAD

**Discussion by Syhkii** [ggml-org/llama.cpp#472] [Q&A]
Weird question maybe, but if i use this ai in one of my coding project cani still sell it ?
Or do i nned to create my own ai for this ?

**Comment by Green-Sky** [comment]
"this ai" - do you mean facebook's llama, or the code at llama.cpp?
- facebook's llama -> no, the model has a [non commercial bespoke license](https://github.com/facebookresearch/llama/blob/main/MODEL_CARD.md)
- llama.cpp -> yes, if you adhere to the conditions in the [MIT license](https://github.com/ggerganov/llama.cpp/blob/master/LICENSE)

**Reply by Kronus86** [reply]
Can you please explain little more. If I understand correctly then llama.cpp should be using parameters (weight, biases etc) from facebook's LLaMA. In that case shouldn't llamap.cpp should also be subjected to the same "non commercial" use.

**Reply by Green-Sky** [reply]
>If I understand correctly then llama.cpp should be using parameters (weight, biases etc) from facebook's LLaMA. In that case shouldn't llamap.cpp should also be subjected to the same "non commercial" use.

If you use them, yes, but llama.cpp itself (the code) is licensed under the MIT license.

**Reply by Kronus86** [reply]
Is there any alternative for the weights. Without weights is there any use of the code?
